packages/hub/src/hub.ts

```diff
@@ -100,413 +102,518 @@ export class Hub implements HubInterface {
    * @param scope bound to the hub.
    * @param version number, higher number means higher priority.
    */
-  public constructor(client?: Client, scope: Scope = new Scope(), private readonly _version: number = API_VERSION) {
-    this.getStackTop().scope = scope;
+  public constructor(client?: Client, scope: Scope = new Scope(), version: number = API_VERSION) {
+    this.version = version;
+    getStackTop(this).scope = scope;
     if (client) {
-      this.bindClient(client);
+      bindClient(this, client);
     }
   }
+}
 
-  /**
-   * @inheritDoc
-   */
-  public isOlderThan(version: number): boolean {
-    return this._version < version;
-  }
-
-  /**
-   * @inheritDoc
-   */
-  public bindClient(client?: Client): void {
-    const top = this.getStackTop();
-    top.client = client;
-    if (client && client.setupIntegrations) {
-      client.setupIntegrations();
-    }
-  }
+/**
+ * Returns the topmost scope layer in the order domain > local > process.
+ *
+ * @hidden
+ * */
+export function getStackTop(hub: Hub): Layer {
+  return hub.stack[hub.stack.length - 1];
+}
 
-  /**
-   * @inheritDoc
-   */
-  public pushScope(): Scope {
-    // We want to clone the content of prev scope
-    const scope = Scope.clone(this.getScope());
-    this.getStack().push({
-      client: this.getClient(),
-      scope,
-    });
-    return scope;
-  }
+/** Returns the scope stack for domains or the process. */
+export function getStack(hub: Hub): Layer[] {
+  return hub.stack;
+}
```

Comment:
A little trick, ultimately in an FP style, having function getters helps since only one-time the `stack` will be present since such key cant be minified type of thing, I think.